### PR TITLE
fix: address uuid missing buffer bounds check

### PR DESCRIPTION
**Pull request by cka121**
### Description

- Bumps `uuid` dependency in `vite-plugin-top-level-await` to the latest version to address a missing buffer bounds check vulnerability.
- Adds `ref` parameter to `actions/checkout` steps in the build workflow to use the PR head SHA during pull request builds, preventing unintended merge commit checkouts.

### Motivation and Context

The `uuid` package had a missing buffer bounds check that could cause undefined behavior. The build workflow checkouts were not pinned to the PR head SHA, which could cause CI to run against the wrong commit. 

### My PR contains...
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,10 @@
     "./packages/*"
   ],
   "overrides": {
-    "serialize-javascript": "^7.0.5"
+    "serialize-javascript": "^7.0.5",
+    "vite-plugin-top-level-await": {
```
Comment (robert-hebel-sb) [inline package.json:51]: this should be fixed by migrating to vite8 and removing vite-plugin-top-level-await package
ref: https://github.com/Menci/vite-plugin-top-level-await/issues/76#issuecomment-4146248796

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -116,3 +116,4 @@ jobs:
       with:
         name: build
         path: ./packages/apidom-playground/build
+
```
Comment (robert-hebel-sb) [inline .github/workflows/build.yml:119]:
```suggestion
```
